filers/default/src/main/java/org/commonjava/indy/filer/def/FileChangeTrackingDecorator.java

```diff
@@ -0,0 +1,153 @@
+import org.commonjava.maven.galley.model.Transfer;
+import org.commonjava.maven.galley.model.TransferOperation;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.TimeZone;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class FileChangeTrackingDecorator
+                extends AbstractTransferDecorator
+{
+    private final Logger logger = getLogger(getClass().getName());
+
+    private final DefaultStorageProviderConfiguration config;
+
+    private volatile int changeCounter;
+
+    private File currentFile;
+
+    private long nextRollDate;
+
+    FileChangeTrackingDecorator( DefaultStorageProviderConfiguration config )
+    {
+        this.config = config;
+        resetTrackingFile();
+    }
+
+    @Override
+    public OutputStream decorateWrite( OutputStream stream, Transfer transfer, TransferOperation op,
+                                       EventMetadata metadata ) throws IOException
+    {
+        return new ChangeTrackingOutputStream( super.decorateWrite( stream, transfer, op, metadata ), transfer );
+    }
+
+    @Override
+    public void decorateCopyFrom( Transfer from, Transfer transfer, EventMetadata metadata ) throws IOException
+    {
+        super.decorateCopyFrom( from, transfer, metadata );
+//        logger.info( "Logging copy-from to changed-file: {}", transfer.getPath() );
+        writeChangedPath( transfer );
+    }
+
+    @Override
+    public void decorateDelete( Transfer transfer, EventMetadata metadata ) throws IOException
+    {
+        super.decorateDelete( transfer, metadata );
+//        logger.info( "Logging delete to changed-file: {}", transfer.getPath() );
+        writeChangedPath( transfer );
+    }
+
+    @Override
+    public void decorateCreateFile( Transfer transfer, EventMetadata metadata ) throws IOException
+    {
+        super.decorateCreateFile( transfer, metadata );
+//        logger.info( "Logging create-file to changed-file: {}", transfer.getPath() );
+        writeChangedPath( transfer );
+    }
+
+    private final synchronized void writeChangedPath( Transfer transfer )
+    {
+        Path storageRoot = config.getStorageRootDirectory().toPath().toAbsolutePath().normalize();
+        Path transferPath = transfer.getDetachedFile().toPath().toAbsolutePath().normalize();
+        if ( transferPath.startsWith( storageRoot ) )
+        {
+            try ( FileWriter fw = new FileWriter( getCurrentListFile(), true ) )
+            {
+//                logger.info( "Change counter: {}", changeCounter );
+                if ( changeCounter > 1 )
+                {
+                    fw.write( "\n" );
+                }
+
+                fw.write( "./" + storageRoot.relativize( transferPath ) );
```

Comment:
@jdcasey here transferPath will have absolute path so i'm not sure what that "./"  (current directory) will add to the absolute path

Reply:
The relativize() call handles that, by making the path relative to the storage root